meetings/2022-05-18.md

```diff
@@ -0,0 +1,120 @@
+* ...: If you have ACP implementation one could also plug it in using componentjs configuration.
+* ...: We will also see sai-aa which is an authorization agent.
+* ...: repos allows to just `npm install` and `npm start`. I will start with Authorization Agent which is main actor in SAI spec. I will demostrate it using my Postman project.
+* ...: The discovery of Agent Registration works by making authenticated HEAD request to Authorization Agent. I'm using dummy tokens in the demo but one can enable solid-oidc via componentsjs config.
+* ...: To access my Agent Reigstration I need to use authenticated request. For unauthenticated request I get 401 with UMA `ticket` and `as_uri`.
+* ...: I can get access token by exchanging my dpop-bound id-token.
+* ...: Now I can request my Agent Registration using the Access Token. And get my Access Grant with all Data Grants which I, as Social Agent have been granted by Alice.
+* ...: From Data Grant I can follow to Data Registration and Data Instances.
+* ...: This also works with other access modes. Demonstrating Create and Delete.
+* ...: Authorization Agent has it's own WebID and acts as agent in solid ecosystem.
+* ...: Authorization Agent is authenticating with Solid-OIDC to get access to user's solid pod.
+* Pavlik: Usually we deal with End-user (`webid`) and the Client/Application (`client-id`).
+* Laurens: Authorizer gets the principle which defines `webid` and `client_id` so one can also get Application Registration.
+* ...: I've changed webid to storage owner's one and set `client_id` and now I get Application Registration
+* Pavlik: I just always try to emphasise the interplay of End-user `webid` and Client/Application `client_id`.
+
+### Solid Roles
+
+https://github.com/solid/authorization-panel/discussions/296
+
+* Pavlik: Shows discussion 296, diagram was expanded from previous version with more end-users and context.
+* ...: On the left we have an end-user, on the right a resource owner (RO). They are typically different social agents from one another (cfr. vision of Solid ecosystem).
+* ...: On the bottom the diagram shows a client application.
+* ...: End-user uses Projectron to use resource owned by RO. Both RO and end-user have WebID, and their own Authorization Agent and OpenID Provider (OP).
+* ...: For clarity the OP of RO was abstracted away.
+* ...: Resource server of End-User contains application registration of end-user. So the end-user is RO of application registration (different context).
+* ...: If we focus on client, it authenticates with an ID Token from the OP. To authorize it uses the end-user's Authorization Agent.
+* ...: Then, if Projectron wants to access a resource on the RS of the RO, upon an unauthenticated request it will get the AS URI and a Permission Ticket. With the AS of the RO it will obtain an access token for the RS.
```

Comment:
```suggestion
* ...: Then, if Projectron wants to access a resource on the RS of the RO, upon an unauthenticated request, it will get the AS URI and a Permission Ticket. With the AS of the RO, it will obtain an access token for the RS.
```